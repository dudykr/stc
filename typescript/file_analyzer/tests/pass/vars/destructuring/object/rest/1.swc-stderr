warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:7:32
   |
7  |   const obj: Readonly<ObjType> = {
   |  ________________________________^
8  | |     foo: 'bar',
9  | |     baz: 'qux',
10 | |     quux: 'quuz',
11 | | }
   | |_^
   |
   = note: {
               foo: 'bar';
               baz: 'qux';
               quux: 'quuz';
           }

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:13:33
   |
13 | export const { foo, ...rest } = obj
   |                                 ^^^
   |
   = note: {
               readonly foo: string;
               readonly baz: string;
               readonly quux: string;
           }

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:15:1
   |
15 | foo
   | ^^^
   |
   = note: string

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:16:1
   |
16 | rest
   | ^^^^
   |
   = note: {
               readonly baz: string;
               readonly quux: string;
           }

