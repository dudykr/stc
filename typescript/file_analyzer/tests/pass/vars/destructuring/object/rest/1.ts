type ObjType = {
    foo: string
    baz: string
    quux: string
}

const obj: Readonly<ObjType> = {
    foo: 'bar',
    baz: 'qux',
    quux: 'quuz',
}

export const { foo, ...rest } = obj
