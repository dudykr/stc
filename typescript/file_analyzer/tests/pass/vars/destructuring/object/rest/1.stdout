warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:7:32
   |
7  |   const obj: Readonly<ObjType> = {
   |  ________________________________^
8  | |     foo: 'bar',
9  | |     baz: 'qux',
10 | |     quux: 'quuz',
11 | | }
   | |_^
   |
   = note: {
               foo: 'bar';
               baz: 'qux';
               quux: 'quuz';
           }

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:13:33
   |
13 | export const { foo, ...rest } = obj
   |                                 ^^^
   |
   = note: {
               foo: string;
               baz: string;
               quux: string;
           }

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:15:1
   |
15 | foo
   | ^^^
   |
   = note: string

warning: Type
  --> $DIR/tests/pass/vars/destructuring/object/rest/1.ts:16:1
   |
16 | rest
   | ^^^^
   |
   = note: {
               baz: string;
               quux: string;
           }

